pillar/vault.py

```diff
@@ -169,6 +169,33 @@ def _authenticate(conn):
         conn.token = os.environ.get('VAULT_TOKEN')
 
 
+def couple(location, conn):
+    """
+    If location is a dictionary, return a dictionary of k->v pairs looked up from vault.
+    If location is a string, return the value looked up from vault.
+    """
+    coupled_data = {}
+    if isinstance(location, basestring):
+        try:
+            (path, key) = location.split('?', 1)
+        except ValueError:
+            (path, key) = (location, None)
+        secret = conn.read(path)
+        if key:
+            secret = secret["data"].get(key, None)
```

Comment:
I think re-using the symbol 'secret' is making this harder to read, maybe find some distinct names?

For example the first could be 'path_data' or something.  I know it has to fan back in to have not have undefined symbols in some branches, but maybe set secret=path_data in the else branch of 'if key'.  Same thinking with the check to see if it is base64 encoded.